src/state/slices/portfolioSlice/selectors.ts

```diff
@@ -637,6 +637,24 @@ export const selectPortfolioTotalBalanceByChainIdIncludeStaking = createCachedSe
   },
 )((_s: ReduxState, filter) => filter?.chainId ?? 'chainId')
 
+export const selectPortfolioTotalBalanceByChainIdIncludeStaking = createDeepEqualOutputSelector(
+  selectPortfolioAccountsUserCurrencyBalancesIncludingStaking,
+  (userCurrencyAccountBalances): Record<ChainId, BigNumber> => {
+    return Object.entries(userCurrencyAccountBalances).reduce<Record<ChainId, BigNumber>>(
+      (acc, [accountId, accountBalanceByAssetId]) => {
+        const chainId = fromAccountId(accountId).chainId
+        if (!acc[chainId]) acc[chainId] = bn(0)
+        Object.values(accountBalanceByAssetId).forEach(assetBalance => {
+          // use the outer accumulator
+          acc[chainId] = acc[chainId].plus(bnOrZero(assetBalance))
+        })
+        return acc
+      },
+      {},
+    )
+  },
+)
+
```

Comment:
Mostly the same implementation as the previous one, except it *actually* indexes by ChainId - renamed the current `selectPortfolioTotalBalanceByChainIdIncludeStaking` to be `selectPortfolioTotalChainIdBalanceIncludeStaking`